usr/lib/pkcs11/common/object.c

```diff
@@ -329,7 +330,8 @@ CK_RV object_flatten(OBJECT * obj, CK_BYTE ** data, CK_ULONG * len)
 
     offset = 0;
 
-    memcpy(buf + offset, &obj->class, sizeof(CK_OBJECT_CLASS_32));
+    class32 = obj->class;
+    memcpy(buf + offset, &class32, sizeof(CK_OBJECT_CLASS_32));
```

Comment:
sizeof(class32) resp. sizeof(*class32) if CK_OBJECT_CLASS_32 is a typedef'ed pointer (i dont hope so).